[Bug] config error: sed: 1: "src/include/pg_config.h": unterminated substitute pattern

### Cloudberry Database version

_No response_

### What happened

    configure: creating ./config.status
    sed: 1: "src/include/pg_config.h": unterminated substitute pattern
    make: *** [config.status] Error 1
    make: *** Deleting file `config.status'

### What you think should happen instead

_No response_

### How to reproduce

config on a non Unix-like OS

### Operating System

Darwin Kernel Version 22.5.0

### Anything else

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://cloudberrydb.org/community/coc).

--- a/configure
+++ b/configure
@@ -24833,4 +24833,11 @@ fi
 # The configure args contain '-Wl,-rpath,\$$ORIGIN`, when it falls
 # as a C literal string, it's invalid, so converting `\` to `\\`
 # to be correct for C program.
-sed -i '/define CONFIGURE_ARGS/s,\([^\\]\)\\\$\$,\1\\\\$$,g' src/include/pg_config.h
+case $build_os in
+darwin*)
+  sed -i '' '/define CONFIGURE_ARGS/s,\([^\\]\)\\\$\$,\1\\\\$$,g' src/include/pg_config.h
+  ;;
+*)
+  sed -i '/define CONFIGURE_ARGS/s,\([^\\]\)\\\$\$,\1\\\\$$,g' src/include/pg_config.h
+  ;;
+esac

